src/main.rs

```diff
@@ -268,6 +268,9 @@ impl RandomDataSource {
 }
 
 impl DataSourceMut for RandomDataSource {
+    fn fetch_description(&mut self) -> DataSourceDescription {
+        DataSourceDescription { source_locator: None }
```

Comment:
report `baseurl` here?

Reply:
For the `RandomDataSource` I think it has to a hard-coded string. We're not necessarily being served out of WASM, so there's no way in general to know where any of this is located, and the profile itself is generated.

Reply:
I interpreted this originally to mean convey None in order to use the hard coded "Legion Prof" but do you mean you think it would be good to display "Random Datasource" or similar in the title?

Reply:
Right, for this `impl` specifically, I would make it `Some("RandomDataSource".to_owned())` (or maybe with spaces instead of camel case), and I'm assuming it gets rendered as `<Source> - Legion Prof` when `Some` and `Legion Prof` or maybe `Unknown Source - Legion Prof` when `None.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/c9504a6e3c2657bf479111cd2b4ae8dd7f2838d6